typescript-cli/build-utilities/precommit.ts

```diff
@@ -0,0 +1,64 @@
+import { fork } from "child_process";
+import projectRoot from "../utilities/projectRoot.js";
+
+const promises = [
+  runESLint(),
+  runJest(),
+  checkTimestamps(),
+];
+
+await Promise.allSettled(promises);
+await Promise.all(promises);
+
+function runESLint() : Promise<void>
```

Comment:
I think that these would be better to run on a CI somewhere, rather than precommit, but that is more preference than anything.

Reply:
There is a conversation on this ongoing further up